Add incident details option for json and sarif formats (#956)
GitGuardian/ggshield: changelog.d/20240906_141336_fnareoh_scrt_4693_ggshield_output_expose_incident_date_and_severity.md

### Added

> - Add `--with-incident-details` to output (json and sarif output only) more information about incident when the secret is already known by the gitguardian dashboard.

Reviewer: ```suggestion
- Add `--with-incident-details` to output more information about known incidents (JSON and SARIF outputs only).
```
